src/audio_device.c

```diff
@@ -447,98 +639,129 @@ inline__ DeviceError write_out(uint32_t device_idx, const int16_t *data, uint32_
         alSourcePlay(device->source);
     }
 
-
-    pthread_mutex_unlock(device->mutex);
+    unlock(output);
     return de_None;
 }
 
-void *thread_poll(void *arg)  // TODO: maybe use thread for every input source
+#ifdef AUDIO
+/* Adapted from qtox,
+ * Copyright © 2014-2019 by The qTox Project Contributors
+ *
+ * return normalized volume of buffer in range 0.0-100.0
+ */
+float volume(int16_t *frame, uint32_t samples)
 {
-    /*
-     * NOTE: We only need to poll input devices for data.
-     */
-    UNUSED_VAR(arg);
-    uint32_t i;
-    int32_t sample = 0;
+    float sumOfSquares = 0;
```

Comment:
sumOfSquares seems to be against the naming standard in toxic. I see it's taken from qtox, but maybe adapt it a bit more to use toxic naming standards.